File: frontend/lib/models/knowledge_base_document_create_dto.dart
class KnowledgeBaseDocumentCreateDto {
  final int clubId;
  final String title;
  final String? description;
  final String? documentType;
  final String? manufacturer;
  final String? equipmentModel;
  final String? language;
  final String? accessLevel;
  final String? fileName;
  final String fileBase64;

  KnowledgeBaseDocumentCreateDto({
    required this.clubId,
    required this.title,
    this.description,
    this.documentType,
    this.manufacturer,
    this.equipmentModel,
    this.language,
    this.accessLevel,
    this.fileName,
    required this.fileBase64,
  });

  Map<String, dynamic> toJson() {
    return {
      'clubId': clubId,
      'title': title,
      'description': description,
      'documentType': documentType,
      'manufacturer': manufacturer,
      'equipmentModel': equipmentModel,
      'language': language,
      'accessLevel': accessLevel,
      'fileName': fileName,
      'fileBase64': fileBase64,
    };
  }
}

File: backend/CoreService/src/main/java/ru/bowling/bowlingapp/DTO/KnowledgeBaseDocumentCreateDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class KnowledgeBaseDocumentCreateDTO {

    @NotNull
    private Long clubId;

    @NotBlank
    private String title;

    private String description;

    private String documentType;

    private String manufacturer;

    private String equipmentModel;

    private String language;

    private String accessLevel;

    private String fileName;

    @NotBlank
    private String fileBase64;
}
